homeassistant/components/keba/sensor.py

```diff
@@ -65,54 +85,41 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
 class KebaSensor(SensorEntity):
     """The entity class for KEBA charging stations sensors."""
 
-    def __init__(self, keba, key, name, entity_type, icon, unit, device_class=None):
+    _attr_should_poll = False
+
+    def __init__(
+        self,
+        keba,
+        entity_type,
+        description: SensorEntityDescription,
+    ):
         """Initialize the KEBA Sensor."""
         self._keba = keba
-        self._key = key
-        self._name = name
+        self._key = description.key
+        self._name = description.name
         self._entity_type = entity_type
-        self._icon = icon
-        self._unit = unit
-        self._device_class = device_class
+
+        self._attr_name = f"{self._keba.device_name} {self._name}"
+        self._attr_unique_id = f"{self._keba.device_id}_{self._entity_type}"
+        self._attr_icon = description.icon
+        self._attr_native_unit_of_measurement = description.native_unit_of_measurement
+        self._attr_device_class = description.device_class
+        self._attr_state_class = description.state_class
+        self._attr_last_reset = description.last_reset
 
         self._state = None
-        self._attributes = {}
+        self._attributes: dict[str, str] = {}
 
     @property
     def should_poll(self):
         """Deactivate polling. Data updated by KebaHandler."""
         return False
 
-    @property
-    def unique_id(self):
-        """Return the unique ID of the binary sensor."""
-        return f"{self._keba.device_id}_{self._entity_type}"
-
-    @property
-    def name(self):
-        """Return the name of the device."""
-        return f"{self._keba.device_name} {self._name}"
-
-    @property
-    def device_class(self):
-        """Return the class of this sensor."""
-        return self._device_class
-
-    @property
-    def icon(self):
-        """Icon to use in the frontend, if any."""
-        return self._icon
-
     @property
     def native_value(self):
```

Comment:
Remove this property and set `self._attr_native_value` in `async_update` instead of `self._state`

Reply:
done